src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -102,17 +105,19 @@ mod imp {
         let addr = siginfo_si_addr(info);
 
         // If the faulting address is within the guard page, then we print a
-        // message saying so.
+        // message saying so and abort.
         if guard != 0 && guard - PAGE_SIZE <= addr && addr < guard {
             report_overflow();
+            libc::raise(libc::SIGABRT);
```

Comment:
Aha, we have a separate signal handler stack here, so I guess that’s a viable way of getting SIGABRT happen. You still need to unregister `SIGABRT` handlers in this process before raising the signal, though.